src/rpm/builder.rs

```diff
@@ -196,18 +153,20 @@ impl RPMBuilder {
         self
     }
 
-    /// Override the automatically-detected name of the build host with a new value
-    pub fn build_host(mut self, build_host: &str) -> Self {
-        self.build_host = build_host.to_owned();
+    pub fn build_host(mut self, build_host: impl AsRef<str>) -> Self {
+        self.build_host = Some(build_host.as_ref().to_owned());
+        self
+    }
+
+    pub fn build_time(mut self, build_time: SystemTime) -> Self {
```

Comment:
Honestly I'm still partial to doing something like this: https://github.com/rpm-rs/rpm/commit/5cfc3be4c67b20f259cce30f41f3a6a057e70b57#diff-b82882f602be2cf2d09fd6546c2097235d9417da83223e9516b8eb783c6257ffR229-R232

We ought to support `SOURCE_DATE_EPOCH`, it's not an RPM specific thing, it's a cross-distro standard for reproducible builds.

https://reproducible-builds.org/docs/source-date-epoch/

Reply:
This is out of scope for the library. It's not a tool, it's a building block/lib that may be used for tools. Providing API to adhere to such specs sure, but not implement any recommendation that's out there.

Reply:
I think that's our main cause of disagreement. I vigorously want to limit scope, enable but not cater Implementation of best practices.

Reply:
For what it's worth, I'm pretty sure rpmlib when used as a library still respects this.  However It's not a hill I will die on